### PR TITLE
Better error handling

### PR DESCRIPTION
**Pull request by aldeed**
- Add ability for `data` argument of `sendResult` to be an `Error`
- Catch handler errors and automatically send a response. Look for `statusCode` and `jsonResponse` properties on thrown errors.
- Add `this.setStatusCode` in methods

Fixes https://github.com/stubailo/meteor-rest/issues/17

### CODE REVIEW

File: packages/rest-accounts-password/rest-login.js [hunk 1]
```diff
@@ -5,60 +5,53 @@ JsonRoutes.add("options", "/users/login", function (req, res) {
 JsonRoutes.add("post", "/users/login", function (req, res) {
```
Comment (aldeed) [inline packages/rest-accounts-password/rest-login.js]: Despite the diff looking complex, the only changes in this file were removing the two try/catches. Errors can just be thrown now.

**Comment by aldeed** [comment]
@stubailo here is the errors development moved to a branch on this repo and squashed to one commit.

**Comment by stubailo** [comment]
@aldeed Thanks, I'll take a look in more detail tomorrow.

**Comment by aldeed** [comment]
https://github.com/stubailo/meteor-rest/pull/39 is against devel branch and replaces this PR.

**Pull request closed without merging**